core/trino-main/src/main/java/io/trino/connector/ConnectorAwareNodeManager.java

```diff
@@ -39,16 +40,15 @@ public ConnectorAwareNodeManager(InternalNodeManager nodeManager, String environ
     @Override
     public Set<Node> getAllNodes()
     {
-        return ImmutableSet.<Node>builder()
-                .addAll(getWorkerNodes())
-                .addAll(nodeManager.getCoordinators())
-                .build();
+        return ImmutableSet.copyOf(nodeManager.getActiveConnectorNodes(catalogName));
     }
 
     @Override
     public Set<Node> getWorkerNodes()
     {
-        return ImmutableSet.copyOf(nodeManager.getActiveConnectorNodes(catalogName));
+        return nodeManager.getActiveConnectorNodes(catalogName).stream()
+                .filter(node -> !node.isCoordinator())
```

Comment:
Ideally this should depend on `node-scheduler.include-coordinator`, otherwise single-node (demo) cluster will not work, will it?

Reply:
Fixed